dali/pipeline/data/tensor_list_test.cc

```diff
@@ -418,23 +457,19 @@ TYPED_TEST(TensorListTest, TestTypeChangeSmaller) {
 
   this->SetupTensorList(&tensor_list, shape, &offsets);
 
-  // Save the pointer
-  const void *ptr = tensor_list.raw_data();
   size_t nbytes = tensor_list.nbytes();
 
   // Change the data type to something smaller
-  tensor_list.template mutable_data<uint8>();
+  tensor_list.template set_type<uint8>();
 
   // Check the internals
   ASSERT_EQ(tensor_list.ntensor(), shape.size());
   for (size_t i = 0; i < tensor_list.ntensor(); ++i) {
+    // ASSERT_EQ(ptrs[i], tensor_list.raw_tensor(i));
     ASSERT_EQ(tensor_list.tensor_shape(i), shape[i]);
     ASSERT_EQ(tensor_list.tensor_offset(i), offsets[i]);
   }
 
-  // No memory allocation should have occured
-  ASSERT_EQ(ptr, tensor_list.raw_data());
```

Comment:
Maybe we can save ptrs to each tensor and then check if no reallocation has happened.

Reply:
I cannot check if no reallocation happen this way, as it will match only for the first pointer. I can add the unsafe call.